enclave-manager/web/packages/app/src/emui/enclaves/components/widgets/RunViaCIButton.tsx

```diff
@@ -0,0 +1,54 @@
+import { Button, ButtonProps, Tooltip } from "@chakra-ui/react";
+import { isDefined } from "kurtosis-ui-components";
+import { useState } from "react";
+import { FiGithub } from "react-icons/fi";
+import { EnclaveFullInfo } from "../../types";
+import { AddGithubActionModal } from "../modals/AddGithubActionModal";
+
+type RunViaCIModalButtonProps = ButtonProps & {
+  enclave: EnclaveFullInfo;
+};
+
+export const AddGithubActionButton = ({ enclave, ...buttonProps }: RunViaCIModalButtonProps) => {
+  const [showModal, setShowModal] = useState(false);
+
+  if (!isDefined(enclave.starlarkRun)) {
+    return (
+      <Button isLoading={true} colorScheme={"yellow"} leftIcon={<FiGithub />} size={"sm"} {...buttonProps}>
+        Add GitHub Action
+      </Button>
+    );
+  }
+
+  if (enclave.starlarkRun.isErr) {
+    return (
+      <Tooltip label={"This enclave didn't really load"}>
```

Comment:
is there any way to display the error or rename to `An error occurred starting the enclave`